applications/ExaquteSandboxApplication/python_scripts/kratos_perturbation_process.py

```diff
@@ -0,0 +1,310 @@
+                    grid_shape = self.correlated_noise_generator["grid_shape"].GetInt(),
+                    ndim=dim)
+            else:
+                self.vector_field_generator = GenerateCN(
+                    corrlen = self.correlated_noise_generator["correlated_length"].GetDouble(),
+                    grid_dimensions = self.correlated_noise_generator["grid_dimensions"].GetVector(),
+                    ndim=dim)
+            if dim == 3:
+                self.mappers = self.Create3DMappers(self.vector_field_generator)
+            else:
+                self.mappers = self.Create2DMappers(self.vector_field_generator)
+
+    def Create3DMappers(self, vector_field_generator):
+        if hasattr(self, 'seed'):
+            vf = vector_field_generator(seed=self.seed)
+        else:
+            vf = vector_field_generator()
+        # normalize generated correlated noise
+        vf = vf/np.linalg.norm(vf)
+        nx, ny, nz = vf.shape[:-1]
+        x_grid = RegularGrid1D(0., self.lx, nx)
+        y_grid = RegularGrid1D(self.y0, self.ly, ny)
+        z_grid = RegularGrid1D(self.z0, self.lz, nz)
+        grid = Grid3D(x_grid, y_grid, z_grid)
+        # mappers become invalid after file_ is destructed
+        mappers = ((KratosMultiphysics.VELOCITY_X, DomainPanel3D(grid, vf[...,0])),
+                   (KratosMultiphysics.VELOCITY_Y, DomainPanel3D(grid, vf[...,1])),
+                   (KratosMultiphysics.VELOCITY_Z, DomainPanel3D(grid, vf[...,2])))
+        return mappers
+
+    def Create2DMappers(self, vector_field_generator):
+        vf = vector_field_generator(seed=self.seed)
+        # normalize generated correlated noise
+        vf = vf/np.linalg.norm(vf)
+        nx, nz = vf.shape[:-1]
+        x_grid = RegularGrid1D(self.x0, self.lx, nx)
+        z_grid = RegularGrid1D(self.z0, self.lz, nz)
+        grid = Grid2D(x_grid, z_grid)
+        # mappers become invalid after file_ is destructed.
+        mappers = ((KratosMultiphysics.VELOCITY_X, DomainPanel2D(grid, vf[...,0])),
+                   (KratosMultiphysics.VELOCITY_Y, DomainPanel2D(grid, vf[...,1])))
+        return mappers
+
+    def ExecuteInitialize(self):
+        self.MapCorrelatedNoiseToModelPart()
+        self.SolveAssociatedPoissonProblem()
+
+    def MapCorrelatedNoiseToModelPart(self):
+        for var, mapper in self.mappers:
+            for node in self.model_part_nodes:
+                vel = mapper.interpolate(node)
+                node.SetSolutionStepValue(var, vel)
+
+    def LoadVelocityField(self):
+        main_model_part_name = self.poisson_parameters["problem_data"]["model_part_name"].GetString()
+        main_model_part = self.model.GetModelPart(main_model_part_name)
+        penalty_coeff = self.poisson_parameters["problem_data"]["penalty_coefficient"].GetDouble()
+        with open(self.poisson_parameters["problem_data"]["load_velocity_field"].GetString()) as dat_file:
```

Comment:
This can also be a future improvement: use hdf5 app or json_output_process from the core to load velocity